src/otx/v2/adapters/torch/dataset.py

```diff
@@ -0,0 +1,210 @@
+"""OTX adapters.torch.mmengine.mmpretrain.Dataset API."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Iterable
+
+import yaml
+from torch.utils.data import DataLoader as TorchDataLoader
+from torch.utils.data import Dataset as TorchDataset
+from torch.utils.data import Sampler
+
+from otx.v2.api.core.dataset import BaseDataset
+from otx.v2.api.utils import set_tuple_constructor
+from otx.v2.api.utils.decorators import add_subset_dataloader
+
+SUBSET_LIST = ["train", "val", "test"]
+
+
+@add_subset_dataloader(SUBSET_LIST)
+class TorchBaseDataset(BaseDataset):
+    """A class representing a dataset for pretraining a model."""
+
+    def _initialize(self) -> None:
+        self.set_datumaro_adapters()  # Set self.dataset_entity & self.label_schema
+        self.initialize = True
+
+    def _build_dataset(
+        self,
+        subset: str,
+        pipeline: list | dict | None = None,
+        config: str | dict | None = None,
```

Comment:
what exactly the difference between `pipeline` and `config`? It is a bit unclear and could be difficult to users to distinguish

Reply:
Yes, I agree that `pipeline` should be renamed to `transforms` in another PR